x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -136,6 +136,7 @@ public PlanFactory visitDissectCommand(EsqlBaseParser.DissectCommandContext ctx)
                 }
                 List<Attribute> keys = parser.outputKeys()
                     .stream()
+                    .filter(x -> x.isEmpty() == false)
                     .map(x -> new ReferenceAttribute(src, x, DataTypes.KEYWORD))
                     .map(Attribute.class::cast)
                     .toList();
```

Comment:
It's really a preference thing - I find the code above more readable and less idiosyncratic:
```
List<Attribute> keys = new ArrayList<>();
for (var x : parser.outputKeys()) {
    if (x.isEmpty() == false) {
         keys.add(new ReferenceAttribute(src, x, DataTypes.KEYWORD));
    }
}
```